JeanneBeula/iReports-back-end
# ft: register user

### What does this PR do?
Display top 8 public user with high score in all languages
### Description of Task to be completed?
- this feature allow creation of new user
- validate user input
- hash the entered password
- and save user info into an array
### How should this be manually tested?
- clone this repo by running `git clone https://github.com/JeanneBeula/iReports-back-end.git`
- run `cd iReports-back-end`
- run `cd code .` to open an editor`
- run `git checkout ft-signup` in order to navigate to this branch
- run `git pull ` in order to have updated code from this branch
- run `git install ` to install all packages
- then run `nodemon index` to start sever
 - after open postman then copy this url `http://localhost:3000/user/signUp` with POST method and  check Body and row then choose JSON format>
 - then provide data rewuired like `{
    "firstName":"Beula",
    "lastName":"Miel",
    "email": "beula@gmail.com",
     "password": "Beula1!"
}` then send request.
- You will get the response of provided data with the hashed password

### Any background context you want to provide?

N/A

### Screenshots


<img width="1418" alt="Screen Shot 2021-07-18 at 13 48 12" src="https://user-images.githubusercontent.com/50265410/126066019-0e734432-7528-41ad-99b8-477a9e9fdc0b.png">
